README.md

```diff
@@ -1 +1,118 @@
-# angular-using-compile-link-readme
\ No newline at end of file
+# Using the compile/link functions
+
+## Overview
+
+We used the `link` function in our previous lab, however we don't actually know much about it. Let's look into `link` and the power it provides us.
+
+## Objectives
+
+- Describe the link function
+- Describe the compile function
+- Describe the Directive's lifecycle
+
+## Directive Lifecycle
+
+Each directive has a lifecycle. Up until now we've only been using one part of the lifecycle - the controller. The controller is initiated and ran just before the directive is about to be mounted into the DOM. This is so we can manipulate any values that we need in time for the rendering.
+
+However, it might come to the point where we need to actually do raw DOM manipulation on the directive itself. One problem - the controller is initiated *before* the directive is in the DOM. What do we do?
```

Comment:
This intro is crucial and I think it's well explained - can you add a detailed example of when you would want to do this? Go step by step through what would happen and why it's important that you manipulate the DOM after the directive is inserted.